frotz.py

```diff
@@ -13,94 +13,99 @@
+from sugar3.graphics.toolbutton import ToolButton
+from sugar3.activity.widgets import EditToolbar, ActivityToolbarButton, StopButton
 
-import vte
 
 class FrotzActivity(activity.Activity):
 
     def __init__(self, handle):
         activity.Activity.__init__(self, handle)
-        
         logging.debug('Starting the Frotz activity')
-        
+
         self.set_title(_('Frotz'))
         self.connect('key-press-event', self.__key_press_cb)
 
-        toolbox = activity.ActivityToolbox(self)
-
-        self._edit_toolbar = activity.EditToolbar()
-        toolbox.add_toolbar(_('Edit'), self._edit_toolbar)
+        self.toolbox = ToolbarBox()
+        self.activity_toolbar_button = ActivityToolbarButton(self)
+        self._edit_toolbar = EditToolbar()
+        button = ToolbarButton()
+        button.set_page(self._edit_toolbar)
+        button.props.icon_name = 'toolbar-edit'
+        button.props.label = _('Edit')
+        self.toolbox.toolbar.insert(button, -1)
+        button.show()
         self._edit_toolbar.show()
+
         self._edit_toolbar.undo.props.visible = False
         self._edit_toolbar.redo.props.visible = False
         self._edit_toolbar.separator.props.visible = False
         self._edit_toolbar.copy.connect('clicked', self._copy_cb)
+        self._edit_toolbar.copy.props.accelerator = '<Ctrl><Shift>C'
         self._edit_toolbar.paste.connect('clicked', self._paste_cb)
+        self._edit_toolbar.paste.props.accelerator = '<Ctrl><Shift>V'
 
-        activity_toolbar = toolbox.get_activity_toolbar()
-        activity_toolbar.share.props.visible = False
-        activity_toolbar.keep.props.visible = False
+        self.toolbox.toolbar.insert(self.activity_toolbar_button, -1)
+        self.activity_toolbar_button.show()
+        self.set_toolbar_box(self.toolbox)
+        activity_toolbar = self.toolbox.get_toolbar()
 
         # Add a button that will send you to the ifarchive to get more games
         activity_toolbar.get_games = ToolButton('activity-get-games')
         activity_toolbar.get_games.set_tooltip(_('Get More Games'))
         activity_toolbar.get_games.connect('clicked', self._get_games_cb)
         activity_toolbar.insert(activity_toolbar.get_games, 2)
         activity_toolbar.get_games.show()
+        self.toolbox.show()
 
-        self.set_toolbox(toolbox)
-        toolbox.show()
-        
-        box = gtk.HBox(False, 4)
+        stop = StopButton(self)
+        self.get_toolbar_box().toolbar.insert(stop, -1)
+        stop.show()
+
+        box = Gtk.HBox(False, 4)
 
         self._vte = VTE()
+        self._vte.set_encoding('utf-8')
         self._vte.show()
         self._vte.connect("child-exited", self._quit_cb)
 
-        scrollbar = gtk.VScrollbar(self._vte.get_adjustment())
+        scrollbar = Gtk.VScrollbar(self._vte.get_vadjustment())
         scrollbar.show()
 
-        box.pack_start(self._vte)
+        box.pack_start(self._vte, True, True, 0)
```

Comment:
providing `kwargs` might make the code more readable

Reply:
Done